### DISCUSSION TITLE
Can Codespaces support to restart the container only?

### DISCUSSION THREAD

**Discussion by luxu-ms** [community/community#10478] [Codespaces]
When the new Codespaces is created, it will start the container and mount to the docker volume to persist the source code.
However, sometimes the container may have no response, while the developer already modified some source code and save. If he remove and create a new Codespaces, the code change on the docker volume may be lost. 

For this scenario, is there any possible that the user can restart the container only and leave the docker volume to persist the source code?

**Comment by asciimike** [comment]
The short answer is that if a developer has edited a file and saved it in the checked out repo, and they restart their codespace, everything is saved.

The slightly longer answer is that if a container fails to start, we will fall back to a recovery container, which will have access to the persisted `/workspaces` directory, but no other containers/volumes that were in your devcontainer.

We recommend saving everything to `/workspaces/**` to ensure the safest behavior as things like `$HOME` or specifically named volumes will be deleted on rebuild or if a codespace falls back to recovery mode.

**Reply by misner** [reply]
@asciimike Hi, I'm running into an issue that might be related to what you said above. I'm trying on CS Code that leverage Github Codesppace and containers (devcontainer, docker-compose...) to persist bash history across container rebuilds. I tried to do what is described here: https://code.visualstudio.com/remote/advancedcontainers/persist-bash-history And for the love of Thor:) I can't do after 10+ hours.

I'm quite new to docker and codespace but I thought that specifically named volumes and mounts like those used in the link above were actually able to persist data across container rebuilds. But you say " specifically named volumes will be deleted on rebuild ". Could you develop a bit what you meant because I'm confused now, but maybe that's why i've failed to achieve what they do on https://code.visualstudio.com/remote/advancedcontainers/persist-bash-history.
